src/js/letters/actions/letters.js

```diff
@@ -37,67 +38,53 @@ export function getLetterList(dispatch) {
     }),
     (response) => {
       window.dataLayer.push({ event: 'letter-list-failure' });
-      if (typeof response.errors === 'undefined' || response.errors.length === 0) {
-        return Promise.reject(new Error('vets_letters_error_server_get: undefined error'));
+      const status = getStatus(response);
+      if (status === '403') {
+        // Backend authentication problem
+        dispatch({ type: BACKEND_AUTHENTICATION_ERROR });
+      } else if (status === '422') {
+        // User has an invalid address for his or her letters
+        dispatch({ type: INVALID_ADDRESS_PROPERTY });
+      } else if (status === '502') {
+        // Some of the partner services are down, so we cannot verify the
+        // eligibility of some letters
+        dispatch({ type: LETTER_ELIGIBILITY_ERROR });
+      } else if (status === '503' || status === '504') {
+        // Either EVSS or a partner service is down or EVSS times out
+        dispatch({ type: BACKEND_SERVICE_ERROR });
+      } else {
+        dispatch({ type: GET_LETTERS_FAILURE });
       }
-      const error = response.errors[0];
-      switch (error.status) {
-        case '503': // Handled same as 504
-        case '504':
-          // Either EVSS or a partner service is down or EVSS times out
-          return dispatch({ type: BACKEND_SERVICE_ERROR });
-        case '403':
-          // Backend authentication problem
-          return dispatch({ type: BACKEND_AUTHENTICATION_ERROR });
-        case '502':
-          // Some of the partner services are down, so we cannot verify the
-          // eligibility of some letters
-          return dispatch({ type: LETTER_ELIGIBILITY_ERROR });
-        default:
-          return Promise.reject(
-            new Error(`vets_letters_error_server_get: ${error.status || 'unknown'}`)
-          );
-      }
-    }
-  ).catch((error) => {
-    if (error.message.match('vets_letters_error_server_get')) {
-      dispatch({ type: GET_LETTERS_FAILURE });
+      throw new Error(`vets_letters_error_getLetterList: ${status}`);
     }
-    throw error;
-  });
+  );
 }
 
-
 export function getBenefitSummaryOptions(dispatch) {
   return apiRequest(
     '/v0/letters/beneficiary',
     null,
-    response => dispatch({
+    (response) => dispatch({
       type: GET_BENEFIT_SUMMARY_OPTIONS_SUCCESS,
       data: response,
     }),
-    () => {
+    (response) => {
       dispatch({ type: GET_BENEFIT_SUMMARY_OPTIONS_FAILURE });
-      // Is there a prize for the longest error name?
-      throw new Error('vets_letters_get_benefit_summary_options_failure');
+      const status = getStatus(response);
+      throw new Error(`vets_letters_error_getBenefitSummaryOptions: ${status}`);
     }
   );
 }
 
-
 // Call getLetterList then getBenefitSummaryOptions
 export function getLetterListAndBSLOptions() {
   return (dispatch) => {
     return getLetterList(dispatch)
-      // Maybe shouldn't try to get BSO options if we get an error or we don't get a BSL...
```

Comment:
Ahh, because of the less convoluted error handling, we don't fetch BSL options if `getLetterList` failed. Brilliant!